flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -331,6 +331,22 @@ public void testListAFileFailsBecauseDirectoryIsExpected() throws IOException {
         FileUtils.listFilesInDirectory(file.toPath(), FileUtils::isJarFile);
     }
 
+    @Test
+    public void testFollowSymbolicLink() throws IOException {
+        final String fileName = "a/a.jar";
+        File folder = tmp.newFolder("a");
+        final File file = tmp.newFile(fileName);
+
+        java.nio.file.Path linkPath = Paths.get(folder.getParent(), "a.lnk");
+        Files.createSymbolicLink(linkPath, file.getParentFile().toPath());
```

Comment:
Instead of jumping back and forth between the folders can we simplify this a bit? For example `File file = new File(folder, "a.jar"); file.createNewFile()`. Or `Files.createSymbolicLink(linkPath, folder.toPath())`.

Reply:
Changed for clearly reading.